newIDE/app/public/External/jsfx/jsfx-main.js

```diff
@@ -0,0 +1,102 @@
+} from '../Utils/pathEditor.js';
+
+const electron = require('electron');
+const ipcRenderer = electron.ipcRenderer;
+const fs = require('fs');
+const remote = electron.remote;
+
+let editorContentDocument,
+  jsfx = null;
+
+const loadMetaData = metaData => {
+  jsfx.CurrentParams = metaData;
+  jsfx.UpdateCurrentView();
+  jsfx.PlayCurrent();
+};
+
+const closeWindow = () => {
+  remote.getCurrentWindow().close();
+};
+
+const saveSoundEffect = (pathEditor) => {
+  jsfx.UpdateDownloadLink(); //Update base64 data
+  let rawData = editorContentDocument.getElementById('download').href; //store data
+  rawData = rawData.replace(/^data:audio\/wav;base64,/, '');
+  fs.writeFile(pathEditor.saveOptions.fullPath, rawData, 'base64', err => {
+    ipcRenderer.send(
+      'jsfx-changes-saved',
+      pathEditor.saveOptions.fullPath,
+      jsfx.CurrentParams
+    );
+    closeWindow();
+  });
+};
+
+// we need to first declare when the window is ready to be initiated
+document.addEventListener('DOMContentLoaded', function () {
+  ipcRenderer.send('jsfx-ready');
+});
+// then trigger bellow from main, this ensures the dom is loaded
+ipcRenderer.on('jsfx-open', (event, receivedOptions) => {
+  const editorFrameEl = document.getElementById('jsfx-frame');
+  jsfx = editorFrameEl.contentWindow;
+  editorContentDocument = editorFrameEl.contentDocument;
+  const presetsPanel = editorContentDocument.getElementById('presets');
+
+  // Load metadata, if it exists
+  if ('jsfx' in receivedOptions.metadata) {
+    loadMetaData(receivedOptions.metadata.jsfx);
+  } else { // If not, simulate a click on the 'Lucky' button to create a random sound effect
+    const generateRandomSoundEffectButton = presetsPanel.childNodes[11];
+    generateRandomSoundEffectButton.click();
+  }
+  // load a custom header
+  const pathEditorHeaderDiv = document.getElementById('path-editor-header');
+  const headerStyle = {
+    saveFolderLabel: 'height:27px;color:SlateGrey;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+    nameInput: 'font-family:"Courier New";height:27px;width:90px;color:SlateGrey;float:left;margin-left: 2px;padding:4px;margin-top: 4px;font-size:15px;border: 2px solid #e5cd50;border-radius: 3px;  ',
+    fileExistsLabel: 'height:27px;color:blue;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+    saveButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    cancelButton: 'height:27px;float:right;margin-right:2px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+  };
+  createPathEditorHeader({
+    parentElement: pathEditorHeaderDiv,
+    editorContentDocument: document,
+    saveToGDFunction: saveSoundEffect,
```

Comment:
nitpicking: rather than calling saveToGDFunction, or xxxFunction, call it onSaveToGD or onXXX.
Reasoning is:
* no need to add "Function" to a variable name, because it's an information about the type that should (in theory) be part of Flow (well we don't flow here but still).
* onXXX conveys the information that this is a function anyway (because it's called "on" some event)
* Finally, it reads like english:

```
onSaveToGD: saveSoundEffect,
onCancelChanges: closeWindow
```

can almost be read: "when saving to GD, please save the sound effect. and when changes are cancelled, please close the window".

(Also, notice how onXXX is generally used for callbacks in React.js... looks familiar? You're passing what are called "props" in React ;))

Reply:
good idea - it is more consistent with the naming conventions in gdevelop5 like that too :D